src/askui/models/shared/settings.py

```diff
@@ -0,0 +1,71 @@
+from pydantic import BaseModel, Field
+
+
+class ChatCompletionsCreateSettings(BaseModel):
+    """
+    Settings for creating chat completions.
+
+    Args:
+        top_p (float | None, optional): An alternative to sampling with temperature,
+            called nucleus sampling, where the model considers the results of the tokens
+            with top_p probability mass. So 0.1 means only the tokens comprising
+            the top 10% probability mass are considered. We generally recommend
+            altering this or `temperature` but not both.
+
+        temperature (float | None, optional): What sampling temperature to use,
+            between 0 and 2. Higher values like 0.8 will make the output more random,
+            while lower values like 0.2 will make it more focused and deterministic.
+            We generally recommend altering this or `top_p` but not both.
+
+        max_tokens (int, optional): The maximum number of tokens that can be generated
```

Comment:
I think we should tell the user what the optional values default to. Also for the other properties.